pkg/reconciler/taskrun/taskrun.go

```diff
@@ -709,6 +709,8 @@ func (c *Reconciler) createPod(ctx context.Context, ts *v1.TaskSpec, tr *v1.Task
 	defer span.End()
 	logger := logging.FromContext(ctx)
 
+	ts = ts.DeepCopy()
```

Comment:
We may want a comment on top explaining why this is here and needed (otherwise, I can definitely see someone else, later, removing it).

Also, any reason to to it here and not above (in what calls `createPod`) ?

Reply:
right, we need a comment then 👼🏼

Reply:
Added the comment.